src/plugins/controls/public/control_group/embeddable/control_group_container.tsx

```diff
@@ -307,6 +307,21 @@ export class ControlGroupContainer extends Container<
     } as ControlPanelState<TEmbeddableInput>;
   }
 
+  protected getReplacementPanelState<TEmbeddableInput extends ControlInput = ControlInput>(
```

Comment:
Just like with `createNewPanelState` above, I needed to separate out this logic so that I could handle secondary consequences as necessary for embeddables of different types - specifically, for controls, I had to handle making sure that the new panel state appeared with the same order as the old panel state.